themes/ddbasic/template.php

```diff
@@ -397,6 +397,25 @@ function ddbasic_preprocess_node(&$variables, $hook) {
 
         $variables['content']['group_right_col_search']['more_link'] = $more_link;
         break;
+
+      case 'ding_page':
+        $more_link = array(
+          '#theme' => 'link',
+          '#text' => t('Read more'),
+          '#path' => 'node/' . $variables['nid'],
+          '#options' => array(
+            'attributes' => array(
+              'title' => $variables['title'],
+            ),
+            'html' => FALSE,
+          ),
+          '#prefix' => '<span class="page-link">',
+          '#surfix' => '</span>',
+          '#weight' => 6,
+        );
+
+        $variables['content']['group_right_col_search']['more_link'] = $more_link;
+        break;
     }
```

Comment:
This is a small one, but you should never use theme link directly. Some preprocessing from the l() function gets lost this way. If you want to output a link with render-array, you should use '#type' => 'link', which adds a pre-render callback that takes care of calling l().

More info: http://drupal.stackexchange.com/questions/24691/how-to-theme-a-link-in-a-d7-render-array#answer-131855